### PR TITLE
Fix `scan dataset` commands not respecting `--local` flag

### LINKED ISSUE #3035: Fides Local Flag not being honored in CI/CD job
### Bug Description

When running a CI/CD example, we aren't able to run it without a hosted fides dependency

#### Steps to Reproduce

1. Try running local flag without a fides instance
2. https://github.com/ethyca/fides-CICD-Quickstart/actions/runs/4670980885/jobs/8271389814

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version:
* OS:
* Python Version:
* Docker Version:

### Additional context

_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes #3035 

### Code Changes

* [x] refactor `local`/`API` command logic at the top-level
* [x] add `local` flags to the dataset scanning functions

### Steps to Confirm

* [ ] `fides --local scan dataset db --credentials-id app_postgres`
* [ ] the command fails connecting to the database (as opposed to a webserver that isn't running)

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] ~~Relevant Follow-Up Issues Created~~
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

Quick fix for a bug that @nicolas-ethyca found

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
This is ready for a review! Not sure what is going on with the test coverage, but I think its ok?

File: src/fides/cli/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from rich_click import secho
+
+
+class LocalModeException(Exception):
+    """Raise an error when `local_mode` is enabled with an incompatible command."""
+
+    def __init__(self, command: str):
```
Comment (ThomasLaPiana) [inline src/fides/cli/exceptions.py:7]: I'm going to add a quick test for this

**Comment by ThomasLaPiana** [comment]
It also looks like some formatting happened here...unintentional

**Review by SteveDMurphy — APPROVED** [review]
Tested and confirmed the new command! Liked the way the implementation was added in there 🙌🏽 

I think you wanted to add another test potentially still but looks good 👍🏽 

re: your formatting comment and some of the developer docs, seems to have become more prevalent lately for sure - maybe we can add something to the docs or CI to help? Might make code reviews less noisy

**Comment by ThomasLaPiana** [comment]
tested locally and verified working

**Comment by nicolas-ethyca** [comment]
Thank you!

**Pull request merged**